js/siteorigin-panels/view/builder.js

```diff
@@ -500,28 +503,18 @@ module.exports = Backbone.View.extend( {
 			defaultPosition: 'first'
 		}, options );
 
-		if ( this.$( '.so-cells .cell' ).length === 0 ) {
-
+		if( ! this.model.get('rows').length ) {
+			// There aren't any rows yet
 			if ( options.createCell ) {
 				// Create a row with a single cell
 				this.model.addRow( [{ weight: 1 }], { noAnimate: true } );
 			} else {
 				return null;
 			}
-
-		}
-
-		var activeCell = this.$( '.so-cells .cell.cell-selected' );
-
-		if ( activeCell.length === 0 ) {
-			if ( options.defaultPosition === 'last' ) {
-				activeCell = this.$( '.so-cells .cell' ).first();
-			} else {
-				activeCell = this.$( '.so-cells .cell' ).last();
-			}
 		}
 
-		return activeCell.data( 'view' ).model;
+		var activeCell = this.activeCell;
+		return _.isEmpty( activeCell ) || this.model.get('rows').indexOf(activeCell.model) === -1 ? this.model.get('rows').last().get('cells').first() : activeCell.model;
```

Comment:
Loooong line is long. :stuck_out_tongue: It's a bit hard to see that the expression

`_.isEmpty( activeCell ) || this.model.get('rows').indexOf(activeCell.model) === -1` 

is what's being checked for the ternary. So perhaps to reduce length of the line and make it more readable extract that expression into a variable before the `return`.